### PR TITLE
TRAIN-984-985-998: create login-register firebase authentication

### PR DESCRIPTION
**Pull request by JaytoRosalijos**
**Things I did**

- implement login/register/logout using firebase authentication
- implement protected routes

[TRAIN-984: Issue Link](https://vananaz.atlassian.net/browse/TRAIN-984?atlOrigin=eyJpIjoiYWQ0MjU4MWQyMjM1NGQ2MTg2ZGIyZGI2OTA2ZWI5NjUiLCJwIjoiaiJ9)
[TRAIN-985: Issue Link](https://vananaz.atlassian.net/browse/TRAIN-985?atlOrigin=eyJpIjoiOTVlMjk5M2QwMGQ0NGRmMGE2MWVmOTJhNWFkOThiZTciLCJwIjoiaiJ9)
[TRAIN-998: Issue Link](https://vananaz.atlassian.net/browse/TRAIN-998?atlOrigin=eyJpIjoiY2Y0ZDNlYmJhYmVjNGYwMzg2YTdmYjAxODBkMDFkODAiLCJwIjoiaiJ9)

### CODE REVIEW

File: src/config/firebase.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { initializeApp } from "firebase/app";
```
Comment (itsmitchyyy) [inline src/config/firebase.tsx:1]: Can you add this fill to .gitignore so you don't expose the api key. @JaytoRosalijos

Comment (JaytoRosalijos) [reply src/config/firebase.tsx:1]: image
is this what you mean for the second option for the api keys? 

should I also inclued the .env file in the .gitignore,?

thanks.

Comment (itsmitchyyy) [reply src/config/firebase.tsx:1]: @JaytoRosalijos yes and also include the .env in .gitignore.

Comment (JaytoRosalijos) [reply src/config/firebase.tsx:1]: Done addressing requested changes @itsmitchyyy 

create .env file for storing firebase keys

**Review by itsmitchyyy — APPROVED** [review]

**Pull request merged**